### PR TITLE
ARROW-12630: [Dev][Integration] conda-integration docker build fails

### PR DESCRIPTION
**Pull request by kszucs**
Disable the rust integration tests by default.

In order to execute the rust integration tests as well pass `ARCHERY_INTEGRATION_WITH_RUST=1` environment variable to the docker container.

### CODE REVIEW

**Comment by pitrou** [comment]
This doesn't seem to be sufficient. I get the following error (multiple times):
```
==========================================================
Testing file /tmp/arrow-integration-_9i6ftux/generated_null_trivial.json
==========================================================
-- Creating binary inputs
-- Validating file
Traceback (most recent call last):
  File "/arrow/dev/archery/archery/integration/runner.py", line 213, in _run_ipc_test_case
    run_binaries(producer, consumer, outcome, test_case)
  File "/arrow/dev/archery/archery/integration/runner.py", line 239, in _produce_consume
    consumer.validate(json_path, producer_file_path)
  File "/arrow/dev/archery/archery/integration/tester_rust.py", line 64, in validate
    return self._run(arrow_path, json_path, 'VALIDATE')
  File "/arrow/dev/archery/archery/integration/tester_rust.py", line 61, in _run
    run_cmd(cmd)
  File "/arrow/dev/archery/archery/integration/util.py", line 139, in run_cmd
    output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
  File "/opt/conda/envs/arrow/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/opt/conda/envs/arrow/lib/python3.8/subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/opt/conda/envs/arrow/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/opt/conda/envs/arrow/lib/python3.8/subprocess.py", line 1706, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/arrow/rust/target/debug/arrow-json-integration-test'
```

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -82,7 +80,7 @@ jobs:
       - name: Setup Archery
         run: pip install -e dev/archery[docker]
       - name: Execute Docker Build
-        run: archery docker run conda-integration
+        run: archery docker run -e ARCHERY_INTEGRATION_WITH_RUST=1 conda-integration
```
Comment (kszucs) [inline .github/workflows/integration.yml:83]: @jorgecarleitao @andygrove @alamb you'll need to update the integration workflow in arrow-rs once we merge this

Comment (jorgecarleitao) [reply .github/workflows/integration.yml:83]: fielded https://github.com/apache/arrow-rs/pull/253

**Comment by kszucs** [comment]
@pitrou verified locally that `archery run conda-integration` works

**Review by pitrou — APPROVED** [review]
+1, thank you @kszucs !

**Pull request closed without merging**